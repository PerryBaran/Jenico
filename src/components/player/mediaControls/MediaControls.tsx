import React, {Dispatch, SetStateAction} from 'react';
import style from './mediaControls.module.css';
import { play, pause, skip } from '../../../media/icons/index';

function MediaControls(props: {playing: boolean, setPlaying: Dispatch<SetStateAction<boolean>>, skipSong: (value?: boolean) => void} ) {
    const { playing, setPlaying, skipSong } = props

    return (
        <div className={style.container}>
            <button onClick={() => skipSong(false)}><img src={skip} alt='skip backward'/></button>
            <button onClick={() => setPlaying(!playing)}><img src={playing? pause : play} alt='play/pause'/></button>
            <button onClick={() => skipSong()}><img src={skip} alt='skip forward'/></button>
        </div>
    );
}

export default MediaControls;