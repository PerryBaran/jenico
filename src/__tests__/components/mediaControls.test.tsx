import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import MediaControls from "../../components/player/mediaControls/MediaControls";

describe("MediaControls", () => {
  const validProps = {
    playing: true,
    handlePlaying: jest.fn(),
    skipSong: jest.fn(),
  };

  afterEach(cleanup);

  test("snapshot", () => {
    const { asFragment } = render(
      <MediaControls
        playing={validProps.playing}
        handlePlaying={validProps.handlePlaying}
        skipSong={validProps.skipSong}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("content, playing === true", () => {
    render(
      <MediaControls
        playing={validProps.playing}
        handlePlaying={validProps.handlePlaying}
        skipSong={validProps.skipSong}
      />
    );

    expect(screen.getAllByRole("button")).toHaveLength(3);
    expect(screen.getByAltText("skip backward")).toHaveAttribute(
      "src",
      "skip.png"
    );
    expect(screen.getByAltText("play/pause")).toHaveAttribute(
      "src",
      "pause.png"
    );
    expect(screen.getByAltText("skip forward")).toHaveAttribute(
      "src",
      "skip.png"
    );
  });

  test("content, playing === false", () => {
    render(
      <MediaControls
        playing={false}
        handlePlaying={validProps.handlePlaying}
        skipSong={validProps.skipSong}
      />
    );

    expect(screen.getByAltText("play/pause")).toHaveAttribute(
      "src",
      "play.png"
    );
  });

  test("buttons", () => {
    render(
      <MediaControls
        playing={false}
        handlePlaying={validProps.handlePlaying}
        skipSong={validProps.skipSong}
      />
    );

    const buttons = screen.getAllByRole("button");
    expect(validProps.skipSong).toBeCalledTimes(0);
    expect(validProps.handlePlaying).toBeCalledTimes(0);
    buttons[0].click();
    expect(validProps.skipSong).lastCalledWith(false);
    buttons[1].click();
    expect(validProps.handlePlaying).lastCalledWith();
    buttons[2].click();
    expect(validProps.skipSong).lastCalledWith();
    expect(validProps.skipSong).toBeCalledTimes(2);
    expect(validProps.handlePlaying).toBeCalledTimes(1);
  });
});
