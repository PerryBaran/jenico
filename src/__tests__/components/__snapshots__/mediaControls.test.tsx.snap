// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaControls snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <button>
      <img
        alt="skip backward"
        src="skip.png"
      />
    </button>
    <button>
      <img
        alt="play/pause"
        src="pause.png"
      />
    </button>
    <button>
      <img
        alt="skip forward"
        src="skip.png"
      />
    </button>
  </div>
</DocumentFragment>
`;
